airbyte-cdk/python/airbyte_cdk/sources/declarative/requesters/http_requester.py

```diff
@@ -76,7 +76,9 @@ def max_retries(self) -> Union[int, None]:
     def retry_factor(self) -> float:
         return self._retrier.retry_factor
 
-    def should_retry(self, response: requests.Response) -> bool:
+    @lru_cache(maxsize=10)
```

Comment:
use a tiny cache to limit the memory footprint. It doesn't have to be large because we mostly only care about the status of the last response received